### PR TITLE
std/http: add serveTLS and listenAndServeTLS

### PR DESCRIPTION
**Pull request by kevinkassimo**
Created `serveTLS` using `listenTLS` for HTTPS instead of `listen` for HTTP.

Example:
```ts
import { serve, serveTLS } from "./std/http/server.ts";

const tlsOptions = {
  hostname: "localhost",
  port: 443,
  certFile: "./localhost.pem",
  keyFile: "./localhost-key.pem",
};

async function httpServer() {
  console.log(`Simple HTTP server listening on localhost:80`);
  for await (const req of serve("localhost:80")) {
    const headers = new Headers();
    headers.set("Location", "https://localhost");
    req.respond({ status: 307, headers });
  }
}

async function httpsServer() {
  console.log(`Simple HTTPS server listening on ${tlsOptions.hostname}:${tlsOptions.port}`);
  const body = new TextEncoder().encode("Hello HTTPS");
  for await (const req of serveTLS(tlsOptions)) {
    req.respond({ body });
  }
}

httpServer();
httpsServer();
```
Screen Shot 2019-11-03 at 11 55 10 AM

There seems to be still some issue when plain HTTP payload is submitted to the `serveTLS` that yields `InvalidData` error. We need to investigate how to gracefully handle the error in potentially another PR.

### CODE REVIEW

File: std/http/server_test.ts [hunk 1]
```diff
@@ -529,4 +529,57 @@ test({
...
+      let serverIsRunning = true;
+      p.status()
+        .then(
+          (): void => {
+            serverIsRunning = false;
+          }
+        )
+        .catch((_): void => {}); // Ignores the error when closing the process.
+
+      await delay(100);
```
Comment (ry) [inline std/http/server_test.ts]: Is this necessary given you wait for "server listening" on stdout above?

File: std/http/server_test.ts [hunk 2]
```diff
@@ -529,4 +529,57 @@ test({
...
+      });
+      await Deno.writeAll(
+        conn,
+        new TextEncoder().encode("GET / HTTP/1.0\r\n\r\n")
+      );
+      const res = new Uint8Array(100);
+      const nread = assertNotEOF(await conn.read(res));
+      conn.close();
+      const resStr = new TextDecoder().decode(res.subarray(0, nread));
+      assert(resStr.includes("Hello HTTPS"));
```
Comment (ry) [inline std/http/server_test.ts]: Nice test.

File: std/http/server.ts [hunk 3]
```diff
@@ -401,6 +401,28 @@ export async function listenAndServe(
   }
 }
 
+export type HTTPSOptions = Omit<Deno.ListenTLSOptions, "transport">;
+
+export function serveTLS(options: HTTPSOptions): Server {
```
Comment (ry) [inline std/http/server.ts]: Can you add a jsdoc with a little example?

File: std/http/server_test.ts (hunk 1 above)
Comment (kevinkassimo) [reply std/http/server_test.ts]: Without this delay we get `Connection refused` error on client side.

File: std/http/server.ts (hunk 3 above)
Comment (kevinkassimo) [reply std/http/server.ts]: Added.

File: std/http/testdata/simple_https_server.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+// This is an example of a https server
+import { serveTLS } from "../server.ts";
+
+const tlsOptions = {
+  hostname: "localhost",
+  port: 4503,
+  certFile: "./http/testdata/tls/localhost.crt",
+  keyFile: "./http/testdata/tls/localhost.key",
+};
+console.log(`Simple HTTPS server listening on ${tlsOptions.hostname}:${tlsOptions.port}`);
```
Comment (ry) [inline std/http/testdata/simple_https_server.ts]: Can you try printing this after serveTLS is called? That might allow removing the delay

**Pull request merged**